File: js/stockPicker.js
exports.picker = function(prices) {

    let initialDifference = prices[1]-prices[0]
    let bestDifference = [initialDifference, 0, 1];  // [diff_value, buyIndex, sellIndex]

    prices.forEach((buy,buyIndex) => {  // buy loop
        prices.forEach((sell,sellIndex) => {  // sell loop
            if (buyIndex < sellIndex && sell-buy > bestDifference[0]) {
                bestDifference = [sell-buy, buyIndex, sellIndex]
            }
        })
    })

    return [bestDifference[1], bestDifference[2]]
}

File: python/stock_picker.py
def picker(prices):
    
    bestDifference = [prices[1]-prices[0], 0, 1] # [best_difference, buy_index, sell_index]
    
    for buy_index in range(len(prices)):
        for sell_index in range(len(prices)):
            if buy_index < sell_index and prices[sell_index]-prices[buy_index] > bestDifference[0]:
                bestDifference = [prices[sell_index]-prices[buy_index], buy_index, sell_index]
                
    return [bestDifference[1], bestDifference[2]]